Objects/unicodeobject.c

```diff
@@ -14083,6 +14083,17 @@ _PyUnicodeWriter_WriteASCIIString(_PyUnicodeWriter *writer,
     return 0;
 }
 
+
+int
+PyUnicodeWriter_WriteASCII(PyUnicodeWriter *writer,
+                           const char *str,
+                           Py_ssize_t size)
+{
+    _PyUnicodeWriter *priv_writer = (_PyUnicodeWriter*)writer;
```

Comment:
It's nice to have sanity-check assertions like this:
```suggestion
    assert(writer != NULL);
    _PyUnicodeWriter *priv_writer = (_PyUnicodeWriter*)writer;
```

If the caller needs a thread state, it would be good to call `_Py_AssertHoldsTstate` as well.

Reply:
Ok, I added the two assertions. Yes, the caller must have an attached tstate.